src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2354,31 +2339,6 @@ private static bool IsThisOrTypeOrNamespace(MemberAccessExpressionSyntax memberA
             return false;
         }
 
-        private static bool IsReplaceableByVar(
```

Comment:
called throuh two codepaths.  one is only called from SimplifyTypeNames.  The other is called through a codepath from simplify type names and by CSharpNameReducer.  So i removed this entirely from the simplification codepaths, but then added into the reduction codepath with a specialized reducer just for this purpose.